### PR TITLE
Mad 8826 affiliate module api documentation[MAD-8826]

### PR DESCRIPTION
**Pull request by Neeraja82**
WIP:Mad 8826 affiliate module api documentation[MAD-8826]

[MAD-8826]: https://liquidweb.atlassian.net/browse/MAD-8826?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Neeraja82** [comment]
Modified and pushed few more changes based on the MAD-8829 code review suggestions

File: affiliate/referral/list.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+**Success Response** (success): 200 OK
+```
+HTTP/1.1 200 OK
+Date: Wed, 03 Nov 2021 12:51:27 GMT
+Content-Type: application/json;charset=utf-8
+Content-Length: 44
+NocWorx-Api-Version: 0.0.0
+
+[
+  {
```
Comment (adrian-enspired) [inline affiliate/referral/list.md:39]: these payloads should look like
```json
[
  {
    "id": 5,
    "identity": "Howell Inc",
    "metadata": {
      "scope": "client",
      "uri": null
    },
    "status": "inactive",
    "referred_date": 1263829324
  },
  . . .
```

File: affiliate/link/list.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+- string `match[status]` (optional): match against status value.
+- string `range[id]` (optional): find system IDs within "{min}..{max}" range
+- integer `page` (optional): 1-based result set count for paginated results.
+- integer `pageSize` (optional): maximum number of results to include per "page" of a paginated list.
+- string `sortBy` (optional): field to sort the list by; one of `id`|`name`|`target_url`|`status`.
+- string `sortOrder` (optional): sort direction; one of `ASC`|`DESC`.
+
+#### Request
+```
+$ curl -i "$PORTAL_API_URL/v1/affiliate/link" \
```
Comment (adrian-enspired) [inline affiliate/link/list.md:26]: for consistency, please make sure we use `$PORTAL_API` and `$PORTAL_KEY` everywhere.

File: affiliate/referral/show.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+**Success Response** (request was accepted for processing): 200 OK
+```
+HTTP/1.1 200 OK
+Date: Wed, 03 Nov 2021 12:51:27 GMT
+Content-Type: application/json;charset=utf-8
+Content-Length: 44
+NocWorx-Api-Version: 0.0.0
+
+{
+  "status": "active",
```
Comment (adrian-enspired) [inline affiliate/referral/show.md:27]: payload should look something like
```json
{
  "id": 36673,
  "identity": "Jacobs, Conn and Steuber",
  "metadata": {
    "scope": "client",
    "uri": null
  },
  "status": "active",
  "referred_date": 1462343023,
  "services": [
    {
      "id": 53721,
      "identity": "M-SIP-100 - 104.207.225.74: beyondgallery.com",
      "metadata": {
        "scope": "service",
        "uri": "/v1/service/shared/53721"
      },
      "status": "enabled",
      "state": "stable",
      "type": "shared"
    }
  ]
}
```

File: affiliate/referral/show.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## affiliate:show
+Show Affiliate Referrals.
+
+#### Access
+affiliate referral view
+
+#### Request
+```
+$ curl -i "$PORTAL_API_URL/v1/affiliate/referral/{id}" \
```
Comment (adrian-enspired) [inline affiliate/referral/show.md:11]: use `$REFERRAL_ID`

File: affiliate/referral/show.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  -H "Authorization: Bearer $PORTAL_API_KEY" \
+  -H "Content-type: application/json" \
+  -H "Accept: application/json"
+```
+
+#### Responses
+**Success Response** (request was accepted for processing): 200 OK
```
Comment (adrian-enspired) [inline affiliate/referral/show.md:18]:
```
**Success Response** (success): 200 OK
```

File: affiliate/link/show.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+#### Request
+```
+$ curl -i "$PORTAL_API_URL/v1/affiliate-link/{id}/stats" \
+  -H "Authorization: Bearer $PORTAL_API_KEY" \
+  -H "Content-type: application/json" \
+  -H "Accept: application/json"
+```
+
+#### Responses
+**Success Response** (request was accepted for processing): 200 OK
```
Comment (adrian-enspired) [inline affiliate/link/show.md:18]:
```
**Success Response** (success): 200 OK
```

File: affiliate/link/show.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+]
+```
+
+**Failure Response** (not logged in, expired token, etc.): 401 Unauthorized
+
+**Failure Response** (insufficient permissions): 403 Forbidden
+
+**Failure Response** (id doesn't exist on your account (or at all)): 404 Not Found
+
+**Failure Response** (other bad input): 422 Unprocessable Entity
```
Comment (adrian-enspired) [inline affiliate/link/show.md:51]: this endpoint won't return 422

File: affiliate/referral/show.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+}"
+```
+
+**Failure Response** (not logged in, expired token, etc.): 401 Unauthorized
+
+**Failure Response** (insufficient permissions): 403 Forbidden
+
+**Failure Response** (id doesn't exist on your account (or at all)): 404 Not Found
+
+**Failure Response** (other bad input): 422 Unprocessable Entity
```
Comment (adrian-enspired) [inline affiliate/referral/show.md:38]: this endpoint won't return 422

**Review by adrian-enspired — CHANGES_REQUESTED** [review]

File: affiliate/referral/list.md (hunk 1 above)
Comment (Neeraja82) [reply affiliate/referral/list.md:39]: Changed, please review. I think shared active status response.

File: affiliate/link/list.md (hunk 2 above)
Comment (Neeraja82) [reply affiliate/link/list.md:26]: As discussed ignoring this, as the changes are not made in the script.

File: affiliate/referral/show.md (hunk 3 above)
Comment (Neeraja82) [reply affiliate/referral/show.md:33]: Changed, please review. I think shared active status response.

File: affiliate/referral/show.md (hunk 4 above)
Comment (Neeraja82) [reply affiliate/referral/show.md]: Changed, please review.

File: affiliate/referral/show.md (hunk 5 above)
Comment (Neeraja82) [reply affiliate/referral/show.md]: Sorry, corrected. Please review.

File: affiliate/link/show.md (hunk 6 above)
Comment (Neeraja82) [reply affiliate/link/show.md]: Sorry, corrected. Please review.

File: affiliate/referral/show.md (hunk 8 above)
Comment (Neeraja82) [reply affiliate/referral/show.md]: Removed, please review.

**Review by adrian-enspired — APPROVED** [review]

**Pull request merged**